src/mqt/core/_core/__init__.pyi

```diff
@@ -1 +1,733 @@
-def add(a: int, b: int) -> int: ...
+from typing import ClassVar, Iterator, overload
+
+class Operation:
+    def __init__(self, arg0: int, arg1: list[Operation], /) -> None: ...
+    def acts_on(self, arg: int, /) -> bool: ...
+    def add_depth_contribution(self, arg: list[int], /) -> None: ...
+    def clone(self) -> Operation: ...
+    @property
+    def controls(self) -> set[Control]: ...
+    @controls.setter
+    def controls(self, arg: set[Control]) -> None: ...
+    def empty(self) -> bool: ...
+    def equals(self, arg0: Operation, arg1: Permutation, arg2: Permutation, /) -> bool: ...
+    def is_classic_controlled_operation(self) -> bool: ...
+    def is_compound_operation(self) -> bool: ...
+    def is_controlled(self) -> bool: ...
+    def is_non_unitary_operation(self) -> bool: ...
+    def is_standard_operation(self) -> bool: ...
+    def is_symbolic_operation(self) -> bool: ...
+    def is_unitary(self) -> bool: ...
+    @property
+    def n_targets(self) -> int: ...
+    @property
+    def n_qubits(self) -> int: ...
+    @property
+    def n_controls(self) -> int: ...
+    @property
+    def name(self) -> str: ...
+    @name.setter
+    def name(self, arg: str) -> None: ...
+    def size(self) -> int: ...
+    @property
+    def targets(self) -> list[int]: ...
+    @targets.setter
+    def targets(self, arg: list[int]) -> None: ...
+    @property
+    def gate(self) -> OpType: ...
+    @gate.setter
+    def gate(self, arg: OpType) -> None: ...
+
+class CompoundOperation(Operation):
+    def __init__(self, arg0: int, arg1: list[Operation], /) -> None: ...
+    def acts_on(self, arg: int, /) -> bool: ...
+    def add_depth_contribution(self, arg: list[int], /) -> None: ...
+    def clone(self) -> Operation: ...
+    @property
+    def controls(self) -> set[Control]: ...
+    @controls.setter
+    def controls(self, arg: set[Control]) -> None: ...
+    def empty(self) -> bool: ...
+    def equals(self, arg0: Operation, arg1: Permutation, arg2: Permutation, /) -> bool: ...
+    def get_starting_qubit(self) -> int: ...
+    def get_used_qubits(self) -> set[int]: ...
+    def is_classic_controlled_operation(self) -> bool: ...
+    def is_compound_operation(self) -> bool: ...
+    def is_controlled(self) -> bool: ...
+    def is_non_unitary_operation(self) -> bool: ...
+    def is_standard_operation(self) -> bool: ...
+    def is_symbolic_operation(self) -> bool: ...
+    def is_unitary(self) -> bool: ...
```

Comment:
Aren't some of these missing the override specifier?

Reply:
There is no @override in python, right? Or do you mean @overload? I don't know too much about annotations in Python but @overload seems to only be used to indicate actual method overloading.